README.md

```diff
@@ -42,6 +42,104 @@ This will construct the platform specific build environment. For example, on
+
+The particular model that is being loaded is an ONNX model that takes two
+placeholders (in the ONNX model sense) named `A` and `B`, and produces another
+output `C`, that is the result of performing a matrix multiplication of those
+two inputs.
+
+```cpp
+FeatureBroker fb(model);
+// Binding inputs and binding outputs.
+auto inputA = fb.BindInput<Tensor<double>>("A:0").value_or(nullptr);
+auto inputB = fb.BindInput<Tensor<double>>("B:0").value_or(nullptr);
+auto output = fb.BindOutput<Tensor<double>>("C:0").value_or(nullptr);
+```
+
+In this particular section we create the `FeatureBroker` *and* associate it
+with a model in the first step. The inputs and outputs are bound, fed, and
+consumed in all the same function so it is simplest to do things that way, but
+in other scenarios it is often common to create a feature broker without any
+model at all, bind inputs to that, and then later on the actual inferencing
+component `.Fork`s another feature broker as a "child" of that feature broker,
+and binds an input. But in this simple example we are simply doing everything
+all at once.
+
+Note again the usage of `tl::expected` for both the input and output pipe
+bindings, since bindings can fail due to a variety of factors. (But in this
+case, they do not, because the inputs and outputs happen to match the types
+advertised by the model.)
+
+```cpp
+auto adata = std::shared_ptr<double>(new double[6]{1, 2, 3, 4, 5, 6});
+auto bdata = std::shared_ptr<double>(new double[12]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12});
+Tensor<double> a(adata, {2, 3});
+Tensor<double> b(bdata, {3, 4});
+inputA->Feed(a);
+inputB->Feed(b);
+Tensor<double> c;
+auto updateExpected = output->UpdateIfChanged(c);
+// This should be an tl::expected<bool, ...>, where there is no error and the value is true.
+const double* cdata = c.Data();
+
+// C should contain the values as indicated in the last value:
+//
+//       / 1 2 3 \ / 1  2  3  4 \   / 38 44  50  56 \
+// A B = \ 4 5 6 / | 5  6  7  8 | = \ 83 98 113 128 /
+//                 \ 9 10 11 12 /
+```
+
+Note that the `Feed` operation here succeeds unconditionally. This is
+important. One of the chief design considerations is that components that feed
+values used in inference did not want to have to deal with failure on account
+of not being what the model expected, or anything else which is practically
+always outside of their control. This is part of the design goal of giving
+feature providing components as little trouble as possible. So long as it is
+the right type, a `Feed` call will succeed. Any bad forms would have to be
+detected and handled downstream by the inferencing component.
+
+Speaking of the inferencing component, let us move on to `UpdateIfChanged`.
+This method returns a `tl::expected<bool, ...>` structure. An error might
+occur for any reason defined by the library. In this case, to give a specific
+example, an error would arise if the bound input matrices were not of
+compatible dimension.)

```

Comment:
wouldnt this error happen when binding the model? It sounds like it would happen during the updateIfChanged.

Reply:
Ah! No it would happen at updateIfChanged as the dimensions are set at feed and are not fixed.

Reply:
Right. Any two dimensional tensors are fine, so long as the number of columns in the first input and number of rows in the second input match. But that doesn't happen at binding, nor at feed, nor really can it given that this operator (like many operators in ONNX) are flexible w.r.t. their inputs size.